### PR TITLE
Introduce new projects configuration and build structure - TEST

### PR DESCRIPTION
**Pull request by Maxez**
## Description

TEST

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: mtd-utils/build.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ if [ $(uname) = "Darwin" ]; then
 	b_log "Building mtd-utils"
 
 	PREFIX_MTD_UTILS=${TOPDIR}/mtd-utils
-	PREFIX_MTD_UTILS_BUILD=${TOPDIR}/_build/host-pc/mtd-utils
+	PREFIX_MTD_UTILS_BUILD=${TOPDIR}/_build/host-generic-pc/mtd-utils
 
 	mkdir -p $PREFIX_MTD_UTILS_BUILD
```
[the remark this answers is not in the document]
Comment (Maxez) [reply mtd-utils/build.sh]: Fixed.

File: mtd-utils/build.sh [hunk 2]
```diff
@@ -24,9 +24,9 @@ if [ $(uname) = "Darwin" ]; then
 
 	make mkfs.jffs2
 	
-	mkdir -p ${TOPDIR}/_build/host-pc/prog
-	mkdir -p ${TOPDIR}/_build/host-pc/prog.stripped
-	cp mkfs.jffs2 ${TOPDIR}/_build/host-pc/prog/
-	cp mkfs.jffs2 ${TOPDIR}/_build/host-pc/prog.stripped/
+	mkdir -p ${TOPDIR}/_build/host-generic-pc/prog
```
[the remark this answers is not in the document]
Comment (Maxez) [reply mtd-utils/build.sh]: Fixed.

File: mtd-utils/build.sh [hunk 3]
```diff
@@ -24,9 +24,9 @@ if [ $(uname) = "Darwin" ]; then
...
+	mkdir -p ${TOPDIR}/_build/host-generic-pc/prog.stripped
```
[the remark this answers is not in the document]
Comment (Maxez) [reply mtd-utils/build.sh]: Fixed.

File: mtd-utils/build.sh [hunk 4]
```diff
@@ -24,9 +24,9 @@ if [ $(uname) = "Darwin" ]; then
...
+	cp mkfs.jffs2 ${TOPDIR}/_build/host-generic-pc/prog/
```
[the remark this answers is not in the document]
Comment (Maxez) [reply mtd-utils/build.sh]: Fixed.

File: mtd-utils/build.sh [hunk 5]
```diff
@@ -24,9 +24,9 @@ if [ $(uname) = "Darwin" ]; then
...
+	cp mkfs.jffs2 ${TOPDIR}/_build/host-generic-pc/prog.stripped/
```
[the remark this answers is not in the document]
Comment (Maxez) [reply mtd-utils/build.sh]: Fixed.

File: mtd-utils/build.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (Maxez) [reply mtd-utils/build.sh]: Fixed.

File: mtd-utils/build.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (Maxez) [reply mtd-utils/build.sh]: Fixed.

File: mtd-utils/build.sh [hunk 6]
```diff
@@ -5,14 +5,14 @@ set -e
 if [ $(uname) = "Darwin" ]; then
```
[the remark this answers is not in the document]
Comment (Maxez) [reply mtd-utils/build.sh]: Fixed.

File: mtd-utils/build.sh [hunk 7]
```diff
@@ -5,14 +5,14 @@ set -e
...
-	PREFIX_MTD_UTILS_BUILD=${TOPDIR}/_build/host-pc/mtd-utils
+	PREFIX_MTD_UTILS="${TOPDIR}/mtd-utils"
+	PREFIX_MTD_UTILS_BUILD="${TOPDIR}/_build/host-generic-pc/mtd-utils"
 
-	mkdir -p $PREFIX_MTD_UTILS_BUILD
+	mkdir -p "$PREFIX_MTD_UTILS_BUILD"
 
-	pushd $PREFIX_MTD_UTILS_BUILD
+	pushd "$PREFIX_MTD_UTILS_BUILD"
 	if [ ! -z "$CLEAN" ]; then
```
[the remark this answers is not in the document]
Comment (Maxez) [reply mtd-utils/build.sh]: Fixed.

**Comment by nalajcie** [comment]
> @nalajcie It wasn't really a whole target family - it was just silently assumed that the ia32-generic is a qemu target etc. Now this project is expected to be stated explicitly. Now we expect a project to deliver a dedicated board_config.h file, so it is simply impossible to build a target without specifying exact project configuration.

I'm not talking about building, but running tests. Previously I could use `trunner.py` for testing on TARGET `armv7m6-imrt106x-besmart1` and now I can't. I know we're in a rush so I'm ok with marking this as TODO/FIXME but it still is what it is - stripping some functionality.

File: .github/workflows/ci-submodule.yml [hunk 8]
```diff
@@ -25,9 +25,9 @@ jobs:
       build_result: ${{ steps.build.outcome }}
     strategy:
       matrix:
-        target: ['armv7a7-imx6ull', 'armv7a9-zynq7000-zedboard', 'armv7a9-zynq7000-qemu', 'armv7m7-imxrt106x', 'armv7m7-imxrt117x', 'armv7m4-stm32l4x6', 'host-pc', 'ia32-generic', 'riscv64-spike', 'riscv64-virt']
+        target: ['armv7a7-imx6ull-evk', 'armv7a9-zynq7000-qemu', 'armv7a9-zynq7000-zedboard', 'armv7a9-zynq7000-zturn', 'armv7m4-stm32l4x6-nucleo', 'armv7m7-imxrt106x-evk', 'armv7m7-imxrt117x-evk', 'host-generic-pc', 'ia32-generic-pc', 'ia32-generic-qemu', 'riscv64-generic-qemu', 'riscv64-generic-spike']
```
[the remark this answers is not in the document]
Comment (Maxez) [reply .github/workflows/ci-submodule.yml:28]: Added separate commit for armv7a9-zynq7000-zturn target.

File: _projects/armv7a9-zynq7000-zturn/board_config.h [hunk 9]
```diff
@@ -0,0 +1,72 @@
+/*
```
[the remark this answers is not in the document]
Comment (Maxez) [reply _projects/armv7a9-zynq7000-zturn/board_config.h:1]: Added separate commit for armv7a9-zynq7000-zturn target.

**Comment by Maxez** [comment]
> Previously I could use `trunner.py` for testing on TARGET `armv7m6-imrt106x-besmart1` and now I can't. I know we're in a rush so I'm ok with marking this as TODO/FIXME but it still is what it is - stripping some functionality.

I've added a suitable `FIXME` comment to `trunner` supported targets definition so that the problem gets tackled in the future.

**Pull request closed without merging**